src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/_docker_utils.py

```diff
@@ -137,29 +141,41 @@ def _get_credentials(cli_ctx,
                      permission=None):
     """Try to get AAD authorization tokens or admin user credentials.
     :param str registry_name: The name of container registry
+    :param str tenant_suffix: The registry login server tenant suffix
     :param str username: The username used to log into the container registry
     :param str password: The password used to log into the container registry
     :param bool only_refresh_token: Whether to ask for only refresh token, or for both refresh and access tokens
     :param str repository: Repository for which the access token is requested
     :param str artifact_repository: Artifact repository for which the access token is requested
     :param str permission: The requested permission on the repository, '*' or 'pull'
     """
+    # Raise an error if password is specified but username isn't
+    if not username and password:
+        raise CLIError('Please also specify username if password is specified.')
+
+    resource_not_found, registry = None, None
     try:
         registry, resource_group_name = get_registry_by_name(cli_ctx, registry_name)
         login_server = registry.login_server
+        if tenant_suffix:
+            logger.warning(
+                "Obtained registry login server '%s' from service. The specified suffix '%s' is ignored.",
+                login_server, tenant_suffix)
     except ResourceNotFound as e:
         # Try to use the pre-defined login server suffix to construct login server from registry name.
         login_server_suffix = get_login_server_suffix(cli_ctx)
         if not login_server_suffix:
             raise
-        registry = None
-        login_server = '{}{}'.format(registry_name, login_server_suffix).lower()
         resource_not_found = str(e)
+        login_server = '{}{}{}'.format(
+            registry_name, '-{}'.format(tenant_suffix) if tenant_suffix else '', login_server_suffix).lower()
 
     # Validate the login server is reachable
     try:
         requests.get('https://' + login_server + '/v2/', verify=(not should_disable_connection_verify()))
     except RequestException:
+        if resource_not_found:
+            logger.warning("%s Default the registry login server to '%s'.", resource_not_found, login_server)
```

Comment:
what does this warning message mean? Tell the user which login server fails to connect? If so maybe rephrase it a bit?

Reply:
Thanks! Just feel "Default the registry login server to xxx" a bit redundant since the connection error always has the login uri. But nit and overall sounds good.

Reply:
I feel the message is still needed since it tells where the possibly incorrect login server is from. The exact wording might be improved though.